sc/social/like/tests/test_upgrades.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+from plone import api
+from sc.social.like.testing import INTEGRATION_TESTING
+
+import unittest
+
+
+class UpgradeTestCaseBase(unittest.TestCase):
+
+    layer = INTEGRATION_TESTING
+
+    def setUp(self, from_version, to_version):
+        self.portal = self.layer['portal']
+        self.setup = self.portal['portal_setup']
+        self.profile_id = u'sc.social.like:default'
+        self.from_version = from_version
+        self.to_version = to_version
+
+    def get_upgrade_step(self, title):
+        """Get the named upgrade step."""
+        self.setup.setLastVersionForProfile(self.profile_id, self.from_version)
+        upgrades = self.setup.listUpgrades(self.profile_id)
+        steps = [s for s in upgrades[0] if s['title'] == title]
+        return steps[0] if steps else None
+
+    def execute_upgrade_step(self, step):
+        """Execute an upgrade step."""
+        request = self.layer['request']
+        request.form['profile_id'] = self.profile_id
+        request.form['upgrades'] = [step['id']]
+        self.setup.manage_doUpgrades(request=request)
+
+    @property
+    def total_steps(self):
+        """Return the number of steps in the upgrade."""
+        self.setup.setLastVersionForProfile(self.profile_id, self.from_version)
+        upgrades = self.setup.listUpgrades(self.profile_id)
+        assert len(upgrades) > 0
+        return len(upgrades[0])
+
+
+class Upgrade1to2TestCase(UpgradeTestCaseBase):
+
+    def setUp(self):
+        UpgradeTestCaseBase.setUp(self, u'3020', u'3030')
+
+    def test_upgrade_to_3030_registrations(self):
+        version = self.setup.getLastVersionForProfile(self.profile_id)[0]
+        self.assertTrue(int(version) >= int(self.to_version))
+        self.assertEqual(self.total_steps, 3)
+
+    def test_update_resource_conditions(self):
+        # check if the upgrade step is registered
+        title = u'Fix caching for WhatsApp'
+        step = self.get_upgrade_step(title)
+        self.assertTrue(step is not None)
```

Comment:
that's because we remove the dependency on unittest2; I think we can drop support for Plone 4.1 now.